src/Core/Validation/FieldValidation/BooleanFieldValidator.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Core\Validation\FieldValidation\FieldValidator;
+
+/**
+ * Validates value is boolean stored as an integer i.e. 1 or 0
+ * true and false are not valid values
+ */
+class BooleanFieldValidator extends FieldValidator
+{
+    protected function validateValue(): ValidationResult
+    {
+        $result = ValidationResult::create();
+        if ($this->value !== true && $this->value !== false) {
```

Comment:
null is not a bool value, so it would be very counterintuitive to have it pass through a boolean validator and say it's valid

I've taken the same approach with the other FieldValidators

null values are instead simply skipped in FieldValidatorsTrait::validate()

DBField has a method nullValue() which returns a default value save to the database if value is null, usually it's 0. In the case of things like DBVarchar, null is an allowable value, so there's no nullValue() method there

Reply:
Have updated so that null values are now allow/skipped in the base FieldValidator class before any of the subclass validateValue() methods are called. This essentially makes BooleanFieldValidator a tri-state validator, though on reflection that's OK because we're validation a field value, rather than a bool, and fields are nullable